domain/consensus/utils/txscript/opcode.go

```diff
@@ -1069,10 +1069,10 @@ func opcodeReturn(op *parsedOpcode, vm *Engine) error {
 }
 
 // verifyLockTime is a helper function used to validate locktimes.
-func verifyLockTime(txLockTime, threshold, lockTime uint64) error {
+func verifyLockTime(txLockTime, threshold, lockTime uint64, isRelativeLockTime bool) error {
 	// The lockTimes in both the script and transaction must be of the same
 	// type.
-	if !((txLockTime < threshold && lockTime < threshold) ||
+	if !isRelativeLockTime && !((txLockTime < threshold && lockTime < threshold) ||
```

Comment:
Why is it needed?

Reply:
because if its relative timelock you don't need to check the types - since there is only one type